include/experimental/__p2642_bits/layout_padded_fwd.hpp

```diff
@@ -79,34 +79,37 @@ struct __is_layout_right_padded_mapping : std::false_type {};
+  std::enable_if_t<std::is_same_v<_Mapping, typename layout_right_padded<_Mapping::padding_value>::template mapping<typename _Mapping::extents_type>>>>
     : std::true_type {};
 
 template <class _LayoutExtentsType, class _PaddedLayoutMappingType>
 constexpr void __check_padded_layout_converting_constructor_mandates()
 {
-  using __extents_type = typename _PaddedLayoutMappingType::extents_type;
-  constexpr auto __padding_stride = _PaddedLayoutMappingType::padding_stride;
-  constexpr auto __idx = __layout_padded_constants<typename _PaddedLayoutMappingType::layout_type, _LayoutExtentsType >::__extent_to_pad_idx;
-  if constexpr ((_LayoutExtentsType::rank() > 1) &&
-                (_LayoutExtentsType::static_extent(__idx) != dynamic_extent) &&
-                (__extents_type::static_extent(__idx) != dynamic_extent) &&
-                (__padding_stride != dynamic_extent)) {
-    if constexpr (__padding_stride == 0) {
-      static_assert(_LayoutExtentsType::static_extent(__idx) == 0);
-    } else {
-      static_assert(
-          _LayoutExtentsType::static_extent(__idx) % __padding_stride == 0);
+  if constexpr (_LayoutExtentsType::rank() > 1) {
+    using __extents_type = typename _PaddedLayoutMappingType::extents_type;
+    constexpr auto __padding_value = _PaddedLayoutMappingType::padding_value;
+    constexpr auto __idx = __layout_padded_constants<typename _PaddedLayoutMappingType::layout_type, _LayoutExtentsType >::__extent_to_pad_idx;
+    if constexpr ((_LayoutExtentsType::static_extent(__idx) != dynamic_extent) &&
+                  (__extents_type::static_extent(__idx) != dynamic_extent) &&
+                  (__padding_value != dynamic_extent)) {
+      if constexpr (__padding_value == 0) {
+        static_assert(_LayoutExtentsType::static_extent(__idx) == 0);
+      } else {
+        static_assert(
+            _LayoutExtentsType::static_extent(__idx) % __padding_value == 0);
+      }
     }
   }
 }
 
 template <typename _ExtentsType, typename _OtherMapping>
 constexpr void __check_padded_layout_converting_constructor_preconditions(const _OtherMapping &__other_mapping) {
-  constexpr auto __padding_stride_idx = __layout_padded_constants<typename _OtherMapping::layout_type, _ExtentsType>::__padding_stride_idx;
-  constexpr auto __extent_to_pad_idx = __layout_padded_constants<typename _OtherMapping::layout_type, _ExtentsType>::__extent_to_pad_idx;
   if constexpr (_ExtentsType::rank() > 1) {
-    assert(__other_mapping.stride(__padding_stride_idx) == __other_mapping.extents().extent(__extent_to_pad_idx));
+    constexpr auto __padded_stride_idx =
+        __layout_padded_constants<typename _OtherMapping::layout_type,
+                                  _ExtentsType>::__padded_stride_idx;
+    constexpr auto __extent_to_pad_idx = __layout_padded_constants<typename _OtherMapping::layout_type, _ExtentsType>::__extent_to_pad_idx;
```

Comment:
`assert` is a macro and disappears in non-debug mode.  Thus, these two new variables will become unused, exposing the code to unused variable warnings.  Would you consider marking them as `[[maybe_unused]]` (which requires C++17, or the equivalent compiler-specific attribute for C++14 mode)?

Reply:
done

Reply:
it is always used, and I have not seen a warning about __extent_to_pad_idx if asserts are not on.